src/protocolMessaging/redux/middleware.js

```diff
@@ -135,13 +136,13 @@ async function getOrderTakerTokenWithQuotes(intent, store, action) {
     console.log(e)
   }
 
-  if (takerTokenBalanceIsZero(store, takerToken)) {
+  if (takerTokenBalanceIsZero(store, action.query.takerToken)) {
     if (maxQuote && BigNumber(takerAmount).gt(maxQuote.takerAmount)) {
       return store.dispatch(gotAlternativeQuoteResponse(maxQuote, action.stackId))
     } else if (quote) {
       return store.dispatch(gotQuoteResponse(quote, action.stackId))
     }
-  } else if (quote && takerTokenBalanceIsLessThanTakerAmount(store, takerToken, quote.takerAmount)) {
+  } else if (quote && takerTokenBalanceIsLessThanTakerAmount(store, action.query.takerToken, quote.takerAmount)) {
```

Comment:
What's the difference between the `takerToken` in `action.query` and in `intent`?

Reply:
@syjk129 `action.query.takerToken` is the token that is actually being used for the order (so in our case `ETH`). `intent.takerToken` is the asset that's being used by the maker (so for a v2 intent, that's `WETH`). When we only had swap v1, these things always matched, but now that we have a wrapper contract, the baseAsset that's being used to fill the order can be different than the baseAsset that the maker has registered on the indexer. The non-alignment of these two values is why balance checks were failing on development before sometimes.